### PR TITLE
Changes to make sure that time bounds are preserved in the correct way.

### PR DESCRIPTION
**Pull request by Katie-Howard**
Addresses #700 

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

I have looked at cube_manipulation.py and changed the functionality of _equalise_coord_bounds slightly. This was doing most of what we wanted it to do, but if there was a cube which didn't have bounds, and one that did, it would get rid of the bounds. This was a historic fix, which after discussion with Stephen Moseley we realised we didn't need. So this functionality has been taken out.

There were unit tests to cover some of the work already, but I have updated and added to the unit tests within test_equalise_coord_bounds. This should run as required. The unit tests now cover these cases (for both the coordinate 'time' and also 'forecast_period':

- Where the cubes have identical bounds
- Where none of the cubes have bounds
- Where the cubes have mismatched bounds (but still values within the bounds)
- Where there is a cube that doesn't have bounds, and a cube that does

### CODE REVIEW

File: lib/improver/tests/utilities/cube_manipulation/test__equalise_coord_bounds.py [hunk 1]
```diff
@@ -99,33 +102,40 @@ def test_with_bounds_threecubes(self):
...
+        cubeB = self.cube.copy()
+        add_forecast_reference_time_and_forecast_period(cubeB)
+        result = iris.cube.CubeList([cubeA, cubeB])
+        _equalise_coord_bounds(result)
+        self.assertTrue(result[0].coord('time').bounds is None)
+        self.assertTrue(result[1].coord('time').bounds is None)
+
+    def test_one_bounds(self):
+        """Test that if there is only 1 set of bounds on the cubes
+        a warning is raised."""
```
Comment (MoseleyS) [inline lib/improver/tests/utilities/cube_manipulation/test__equalise_coord_bounds.py]:
```suggestion
        an error is raised."""
```

File: lib/improver/tests/utilities/cube_manipulation/test__equalise_coord_bounds.py [hunk 2]
```diff
@@ -99,33 +102,40 @@ def test_with_bounds_threecubes(self):
...
+        msg = "Cubes with mismatching bounds are not compatible"
+        with self.assertRaisesRegex(ValueError, msg):
+            _equalise_coord_bounds(result)
 
-    @ManageWarnings(record=True)
-    def test_remove_bounds_threecubes(self, warning_list=None):
-        """Test that the second of three input cube has bounds removed and
-        warning is raised."""
+    def test_one_bounds_threecubes(self):
+        """Test that when one of the input cubes has no bounds a warning is
```
Comment (MoseleyS) [inline lib/improver/tests/utilities/cube_manipulation/test__equalise_coord_bounds.py]:
```suggestion
        """Test that when one of the input cubes has no bounds an error is
```

File: lib/improver/tests/utilities/cube_manipulation/test__equalise_coord_bounds.py [hunk 3]
```diff
@@ -138,6 +148,61 @@ def test_mismatched_bounds(self):
...
+        cubeB = self.cube.copy()
+        add_forecast_reference_time_and_forecast_period(cubeB)
+        bounds = [0., 1.]
+        cubeA.coord('forecast_period').bounds = bounds
+        cubeB.coord('forecast_period').bounds = bounds
+        result = iris.cube.CubeList([cubeA, cubeB])
+        _equalise_coord_bounds(result)
+        self.assertArrayAlmostEqual(result[0].coord('forecast_period').bounds,
+                                    [bounds])
+        self.assertArrayAlmostEqual(result[1].coord('forecast_period').bounds,
```
Comment (MoseleyS) [inline lib/improver/tests/utilities/cube_manipulation/test__equalise_coord_bounds.py]: If we are expecting the values to be literally unchanged, then we could use assertArrayEqual instead.

File: lib/improver/utilities/cube_manipulation.py [hunk 4]
```diff
@@ -444,9 +444,7 @@ def _equalise_coord_bounds(cubes):
     """
     # Check each cube against all remaining cubes
     def warn(cube):
-        """Raise warning about  mismatched bounds"""
-        warnings.warn('Removing mismatched bounds from cube {}'.format(
-            cube.name))
+        """Raise warning about mismatched bounds"""
```
Comment (MoseleyS) [inline lib/improver/utilities/cube_manipulation.py]: I think this function isn't being used any more and should be deleted.

File: lib/improver/tests/utilities/cube_manipulation/test__equalise_coord_bounds.py (hunk 1 above)
Comment (Katie-Howard) [reply lib/improver/tests/utilities/cube_manipulation/test__equalise_coord_bounds.py]: Done!

File: lib/improver/tests/utilities/cube_manipulation/test__equalise_coord_bounds.py (hunk 2 above)
Comment (Katie-Howard) [reply lib/improver/tests/utilities/cube_manipulation/test__equalise_coord_bounds.py]: Done.

**Comment by Katie-Howard** [comment]
I have made the small changes that Stephen asked me to make, so it should be ready for a second review now.

**Pull request merged**